midend/simplifySelectCases.cpp

```diff
@@ -42,6 +42,15 @@ void DoSimplifySelectCases::checkSimpleConstant(const IR::Expression* expr) cons
     auto ei = EnumInstance::resolve(expr, typeMap);
     if (ei != nullptr)
         return;
+
+    // we allow value_set name to be used in place of select case;
+    if (expr->is<IR::PathExpression>()) {
```

Comment:
You cannot hardwire the name of an extern block in the mid-end!
The name of this block is architecture-specific!

Reply:
The spec says "Some targets may require that all keyset expressions in a select expression be compile-time known values." In this case, the mid-end should not check the selection cases are all compile-time constant, because it is target-specific, thus should be done in the backend.

Reply:
You cannot change the midend files to depend on specifics of a particular architecture; you can change the midend.cpp file in the bmv2 back-end, which assembles the mid-end files that it needs.
If the mid-end has architecture-specific choices, these should be encapsulated in an abstract policy method which is defined in the back-end.

Reply:
The right approach is probably to have a policy object that encapsulates the set of extern types that can appear in a select case.  The backend would create a policy containing value_set and provide it to this pass.  Bckends without value_set or anything similar would provide an empty policy.